sign/eddsa/eddsa.go

```diff
@@ -69,6 +73,9 @@ func (e *EdDSA) UnmarshalBinary(buff []byte) error {
 	e.seed = buff[:32]
 	e.prefix = prefix
 	e.Secret = secret
+	if e.Secret.Equal(group.Scalar().Zero()) {
+		return errors.New("invalid secret key")
```

Comment:
Again, I'm not sure if the test should be here... Do you really want to make sure that you cannot unmarshal a zero? Perhaps as it's only for EdDSA it might be OK.